de-lib/src/main/java/org/iplantc/de/commons/client/validators/ImageNameValidator.java

```diff
@@ -0,0 +1,44 @@
+package org.iplantc.de.commons.client.validators;
+
+import org.iplantc.de.resources.client.messages.IplantValidationMessages;
+
+import com.google.common.base.Strings;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.editor.client.Editor;
+import com.google.gwt.editor.client.EditorError;
+
+import com.sencha.gxt.widget.core.client.form.error.DefaultEditorError;
+import com.sencha.gxt.widget.core.client.form.validator.AbstractValidator;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by sriram on 6/9/17.
+ */
+public class ImageNameValidator extends AbstractValidator<String> implements IPlantDefaultValidator {
+
+    private final IplantValidationMessages validationMessages;
+
+    public ImageNameValidator() {
+        this(GWT.create(IplantValidationMessages.class));
+    }
+
+    ImageNameValidator(IplantValidationMessages validationMessages) {
+        this.validationMessages = validationMessages;
+    }
+
+    @Override
+    public List<EditorError> validate(Editor<String> editor, String value) {
+        if (Strings.isNullOrEmpty(value)) {
+            return createError(new DefaultEditorError(editor, validationMessages.imgNameReqd(), value));
+        }
+
+        if (value.matches("^[\\s/:].*|.*[\\s/:]$")) {
```

Comment:
I believe in ToolAdminDetailsDialog we're adding `docker.cyverse.org/` to the tool image value.  Is that because we only ever want the images to start with `docker.cyverse.org`? If so, do we need some sort of validation to make sure they didn't remove `docker.cyverse.org` and put in a different URL?